### PR TITLE
fix(lambda): update default runtimes and tests to node 16 everywhere

### PR DESCRIPTION
**Pull request by mrgrain**
Upgrades a few runtime versions from `NODEJS_14_X` (which is EOL and will enter Deprecation Phase 1 on Nov 27, 2023) to `NODEJS_16_X`.

Also updates all the tests.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-iam/lib/oidc-provider/external.ts [hunk 1]
```diff
@@ -1,11 +1,4 @@
 /* istanbul ignore file */
-// the X509 certificate API is available only in node16.
-// since we compile the repo against node 14, typechecking it will fail.
-// its currently too complex to configure node16 only on this
-// file (jsii doesn't support custom tsconfig)
-// so we disable typechecking. don't worry, we have sufficient integ tests that
-// validate this code doesn't break.
-// @ts-nocheck
```
Comment (mrgrain) [inline packages/aws-cdk-lib/aws-iam/lib/oidc-provider/external.ts:8]: New jsii version works with node16

File: packages/aws-cdk-lib/aws-lambda-nodejs/lib/bundling.ts
Comment (mrgrain) [inline packages/aws-cdk-lib/aws-lambda-nodejs/lib/bundling.ts:1]: This file has non test changes

File: packages/aws-cdk-lib/aws-iam/lib/oidc-provider/external.ts
Comment (mrgrain) [inline packages/aws-cdk-lib/aws-iam/lib/oidc-provider/external.ts:1]: This file has non test changes

File: packages/aws-cdk-lib/aws-events-targets/lib/aws-api.ts
Comment (mrgrain) [inline packages/aws-cdk-lib/aws-events-targets/lib/aws-api.ts:1]: This file has non test changes

File: packages/aws-cdk-lib/aws-eks/lib/cluster.ts
Comment (mrgrain) [inline packages/aws-cdk-lib/aws-eks/lib/cluster.ts:1]: This file has non test changes

File: packages/aws-cdk-lib/aws-ses/lib/receipt-rule.ts
Comment (mrgrain) [inline packages/aws-cdk-lib/aws-ses/lib/receipt-rule.ts:1]: This file has non test changes

**Comment by mrgrain** [comment]
SDKv3 todo:

https://github.com/aws/aws-cdk/blob/a523e9fc81021897285b656c49c1e71d888a38f7/packages/aws-cdk-lib/aws-events-targets/lib/aws-api-handler/index.ts

**Review by rix0rrr — APPROVED** [review]

File: packages/aws-cdk-lib/aws-events-targets/lib/aws-api.ts [hunk 2]
```diff
@@ -86,7 +86,7 @@ export class AwsApi implements events.IRuleTarget {
       code: lambda.Code.fromAsset(path.join(__dirname, 'aws-api-handler'), {
         exclude: ['*.ts'],
       }),
-      runtime: lambda.Runtime.NODEJS_14_X,
+      runtime: lambda.Runtime.NODEJS_16_X,
```
Comment (mrgrain) [inline packages/aws-cdk-lib/aws-events-targets/lib/aws-api.ts:89]: Only to node16 because it is still using SDKv2

**Pull request merged**